tests/queries/0_stateless/02763_last_day_of_week.sql

```diff
@@ -0,0 +1,17 @@
+WITH 
```

Comment:
Partially done. I had integrated this test into existing test file `00941_to_custom_week.sql`.
After that, in order to avoid series of unwanted commits, I was contemplating the grepped tests for a while and came across some structure, of course - as well as a fair bit of a mess.
I looked at the test families of the 3 existing functions - `toStartOfMonth`, `toStartOfWeek` and `toLastDayOfMonth` - and grouped my observations into a table below.
The headers are the names of the functions being considered.
The first part of the table shows that `toStartOfWeek` and `toLastDayOfWeek` have their own week-related test and `toStartOfWeek` is also tested in its _own_ `02176_toStartOfWeek_overflow_pruning` test.
In order to save some space, I didn't include the `toLastDayOfMonth` function in the table, but it is tested _only_ in two tests - its _own_ test `02251_last_day_of_month` and a template test generator `02403_enable_extended_results_for_datetime_functions.sql.j2` (which presumably now includes all `toStartOf*`/`toLastDayOf` functions).
The middle part of the table shows a group of tests which were probably meant to include all the observed functions, but actually some include only a part of them.
And the bottom part shows the grepped tests that presumably do not test the functions as such but merely use them as auxiliary tools.
The last column contains my suggestions and doubts regarding the new `toLastDayOfWeek` function.

| toStartOfMonth                                       | toStartOfWeek                                        | toLastDayOfWeek                                      |
|------------------------------------------------------|------------------------------------------------------|------------------------------------------------------|
|                                                      | 00941_to_custom_week                                 | 00941_to_custom_week (already integrated here)       |
|                                                      |                                                      |                                                      |
|                                                      | 02176_toStartOfWeek_overflow_pruning                 | (should I create similar test here?)                 |
|------------------------------------------------------|------------------------------------------------------|---------------------------------------------------   |
| 00189_time_zones_long                                | (absent, but should have also been included?)        | (ignore this test or include all 3 (4?) functions?)  |
| 01472_toStartOfInterval_disallow_empty_tz_field      | (absent, but should have also been included?)        | (ignore this test or include all 3 (4?) functions?)  |
| 01921_datatype_date32                                | 01921_datatype_date32                                | 01921_datatype_date32                                |
| (absent, but should have also been included?)        | 01958_partial_hour_timezone                          | (ignore this test or include all 3 (4?) functions?)  |
| 02096_date_time_1970_saturation                      | 02096_date_time_1970_saturation                      | 02096_date_time_1970_saturation                      |
| 02096_date_time_1970_saturation2                     | 02096_date_time_1970_saturation2                     | 02096_date_time_1970_saturation2                     |
| 02112_with_fill_interval                             | 02112_with_fill_interval                             | (not quite sure if `toLastDayOfWeek` fits here)      |
| 02403_enable_extended_results_for_datetime_functions | 02403_enable_extended_results_for_datetime_functions | 02403_enable_extended_res..(already integrated here) |
|------------------------------------------------------|------------------------------------------------------|---------------------------------------------------   |
| 00141_parse_timestamp_as_datetime                    | 02540_duplicate_primary_key                          |                                                      |
| 01515_mv_and_array_join_optimisation_bag             | 02540_duplicate_primary_key2                         |                                                      |
| 01622_multiple_ttls                                  |                                                      |                                                      |
| 02014_storage_merge_order_by                         |                                                      |                                                      |
| 02149_read_in_order_fixed_prefix                     |                                                      |                                                      |
| 02271_fix_column_matcher_and_column_transformer      |                                                      |                                                      |
| 00054_merge_tree_partitions                          |                                                      |                                                      |

Could you please take a look and share your thoughts on what existing (or even new) tests the `toLastDayOfWeek` function is most appropriate for?

Reply:
Interestingly, while working on `00189_time_zones_long` test, I noticed what could be a hasty copy-paste from the adjacent test case, namely query repetition. Here how it looks and how I suggest to solve:

![image](https://github.com/ClickHouse/ClickHouse/assets/29695184/9a4a1605-0fc3-4475-9638-285b722b0d41)

And this relates not only to `toMonday()` function, but also at least to `toStartOfMonth`, `toStartOfQuarter` and `toStartOfYear`.
Should I fix it in the current pull request or is it a matter of a separate one?

Reply:
A separate PR would be the cleanest approach but 1. CI is quite badly under load and 2. the proposed fix is straightforward. So let's correct that in this PR, thanks.

Reply:
Is it OK if I rename `01472_toStartOfInterval_disallow_empty_tz_field` test to `01472_toBoundsOfInterval_disallow_empty_tz_field` after integrating `toLastDayOfWeek`?
Especially considering that some day we might decide to integrate `toLastDayOfMonth` or even some new `toLastDayOf*` in this test.

Reply:
Almost done. Is it OK if I rename `01472_toStartOfInterval_disallow_empty_tz_field` test to `01472_toBoundsOfInterval_disallow_empty_tz_field` after integrating `toLastDayOfWeek`?
Especially considering that some day we might decide to integrate `toLastDayOfMonth` or even some new `toLastDayOf*` in this test.

Reply:
Sure. Thank you!
The two files have already been renamed, I just try not to spam commits too often, in order to mitigate the CI load. 
The final commit is on the way :)

Reply:
Done.